examples/vision/reptile.py

```diff
@@ -0,0 +1,248 @@
+        # and put that data into a dictionary.
+        self.data = {}
+
+        def extraction(image, label):
+            # This function will shrink the omniglot images into the desired size,
+            # scale the pixel values and convert the RGB image to graysclae.
+            image = tf.image.convert_image_dtype(image, tf.float32)
+            image = tf.image.rgb_to_grayscale(image)
+            image = tf.image.resize(image, [28, 28])
+            return image, label
+
+        for image, label in ds.map(extraction):
+            image = image.numpy()[0]
+            label = str(label.numpy()[0])
+            if label not in self.data:
+                self.data[label] = []
+            self.data[label].append(image)
+            self.labels = list(self.data.keys())
+
+    def get_mini_dataset(
+        self, batch_size, repetitions, shots, num_classes, split=False
+    ):
+
+        temp_labels = np.zeros(shape=(num_classes * shots))
+        temp_images = np.zeros(shape=(num_classes * shots, 28, 28, 1))
+        if split:
+            test_labels = np.zeros(shape=(num_classes))
+            test_images = np.zeros(shape=(num_classes, 28, 28, 1))
+        # Get a random subset of labels from the entire label set.
+        label_subset = random.choices(self.labels, k=num_classes)
+        for class_idx, class_obj in enumerate(label_subset):
+            # Use enumerated index value as a temporary label for mini_batch in
+            # few shot learning.
+            temp_labels[class_idx * shots : (class_idx + 1) * shots] = class_idx
+            # If creating a split dataset for testing, select an extra sample from each
+            # label to create the test dataset.
+            if split:
+                test_labels[class_idx] = class_idx
+                images_to_split = random.choices(
+                    self.data[label_subset[class_idx]], k=shots + 1
+                )
+                test_images[class_idx] = images_to_split[-1]
+                temp_images[
+                    class_idx * shots : (class_idx + 1) * shots
+                ] = images_to_split[:-1]
+            else:
+                # For each index in the randomly selected label_subset, sample the
+                # necessary number of images.
+                temp_images[
+                    class_idx * shots : (class_idx + 1) * shots
+                ] = random.choices(self.data[label_subset[class_idx]], k=shots)
+
+        dataset = tf.data.Dataset.from_tensor_slices(
+            (temp_images.astype(np.float32), temp_labels.astype(np.int32))
+        )
+        dataset = dataset.shuffle(100).batch(batch_size).repeat(repetitions)
+        if split:
+            return dataset, test_images, test_labels
+        return dataset
+
+
+import urllib3
+
+urllib3.disable_warnings()
+train_dataset = Dataset(training=True)
+test_dataset = Dataset(training=False)
+
+"""
+## Build the Model
+"""
+
+
+class ModelLayer(layers.Layer):
```

Comment:
This layer is a simple combination of built-in layers, so prefer structuring it in a functional way:

```python
def conv_bn(x):
   x = layers.Conv2D(filters=64, kernel_size=3, strides=2, padding="same")(x)
   x = layers.BatchNormalization()(x)
   return layers.ReLU()(x)
```